lib/ferrum/network.rb

```diff
@@ -62,7 +62,14 @@ def wait_for_idle(connections: 0, duration: 0.05, timeout: @page.browser.timeout
       start = Utils::ElapsedTime.monotonic_time
 
       until idle?(connections)
-        raise TimeoutError if Utils::ElapsedTime.timeout?(start, timeout)
+        if Utils::ElapsedTime.timeout?(start, timeout)
+          if @page.browser.options.pending_connection_errors
+            pendings = traffic.select(&:pending?)
+            pending_urls = pendings.map(&:url).compact
```

Comment:
Does the pending connection always have a url here? I thought we saw that sometimes it was empty, because the pendings are a collection of Network Exchange objects, from which the url method is `request&.url`. I could be wrong here - and it could also be that when we tested last time the exchange didn't have a request (and therefore no url) but when it's run normally they do.